### ISSUE TITLE
Diversifying fates testmods included in `aux_clm` test suite

### ISSUE THREAD

**Issue by glemieux** [ESCOMP/CTSM#1265]
Per our [discussion in the software meeting today](https://github.com/ESCOMP/CTSM/wiki/Meeting-Notes-2021-Software#fates-merge-into-ctsm-master), this issue is meant to provide us a place to discuss the diversification of fates testmods in the `aux_clm` test suite.  With the upcoming merging of fates_main_api into ctsm master (#1264) and all future development happening directly off of master, I'm thinking that we want to have more coverage of a variety of fates testmods.  My concern is that with `aux_clm` only using `FatesColdDef` for the fates testmod, we might miss an answer changing result due to a ctsm-side changes (i.e. not initiated by fates api updates) for different fates configurations.  I'm not sure how likely this is to be though.  

Here's the questions that I have:

1. Based on @billsacks suggestion, what subset of `aux_clm` fates tests should we _replace_ with tests from `fates` suite.  The idea being that we want to minimize the number of new tests added to `aux_clm`.
2. What compset coverage is necessary?  The `fates` suite predominantly tests using `I2000Clm50FatesCru` where as `aux_clm` fates test uses `I2000ClmXXFates`, where `XX` are the different clm versions.  Similarly, should we diversify the compset coverage in the `fates` suite?
3.  From a process standpoint, if a ctsm-side change does create answer changes to fates, how do we want to handle assessing the such an update?  My assumption is that given fates team attendance to the ctsm software meeting, we will not likely miss out on fates answer-changing updates and will discuss stuff there and here on github.

**Comment by glemieux** [comment]
Oh a couple other related things:

1. Should we discuss #514 within this scope?
2. Should we add a edit the "Fates next api` to strike all mention of "next" or create a new tag for fates api changes?

**Comment by ekluzek** [comment]
It does seem like you should at least have a NCK test in the fates test suite (addressing #514).

And yes we should change or remove the labels. Do we need them anymore? The ChangeLog will document about the update to the FATES version. Part of the reason for them was to point out that a PR had a FATES API update that then required a HLM update. Maybe only that label is useful now?

**Comment by ekluzek** [comment]
Note, I've talked to @glemieux about rerunning testing in order to get all the code review changes in. If that's done it's likely this could be part of PR #1264. But, it's fine either way.

**Comment by ekluzek** [comment]
In terms of what to test. I think we still want to have two test lists, one for FATES and one for standard CTSM (aux_clm). The FATES test list should exhaustively test all the FATES features that the FATES community wants to keep working each tag. The aux_clm test list should have a few FATES tests -- but geared towards the ones that are most likely to be used by the entire CTSM community. Since, FATES has been considered a optional development aspect for a smaller community hat means just a simple FATES test was probably OK. Now, that FATES is being considered more of a main-line development (although still not the standard one) we'll want to increase the FATES testing. But, I still think that aux_clm should test only the most mature, most used, and most standard FATES options that the broader CTSM community is most likely to use.

Because the FATES test list will be run by FATES developers for FATES tags, this is probably OK. That likely means not all tags will have the FATES test list run on them, and as such FATES developers might also have to run the FATES test list for the baseline code as well before making a tag. It's possible we'll then see that something gets broken in FATES between tags where the full FATES test list is run. But, my guess is that won't be that often and it won't be much of a problem. That does mean FATES should likely recommend only the CTSM tags that had the full FATES test suite run on them to FATES developers though.

**Comment by glemieux** [comment]
> And yes we should change or remove the labels. Do we need them anymore? The ChangeLog will document about the update to the FATES version. Part of the reason for them was to point out that a PR had a FATES API update that then required a HLM update. Maybe only that label is useful now?

I find the labels helpful for sorting through the PRs quickly.  I think `FATES API` is probably the most important.  That said, if a PR is only updating the parameter file for fates and/or updating the externals pointer to a new version of fates (i.e. no real update to ctsm-side code), I could see using a `FATES change` tag or something similar.

**Comment by ekluzek** [comment]
We talked about which FATES test-mods shouldn't be in the aux_clm test list and these are the ones we came up with: FatesNoFire, FatesPPhys, FatesShortList, FatesPRT2 (until changes start to come in on the CTSM side to support it), FatesST3. It looks like FatesFire can just be removed as it's the default.

**Comment by ekluzek** [comment]
I also suggest that the Fates base class test-mod should have the COLDSTART setting removed. But, then all other current tests should inherent from the FatesColdDef class which does a COLDSTART, and the names of them all should have FatestColdDef in the name.

**Comment by ekluzek** [comment]
We also think that some tests that start from inventories need to be setup so we aren't always doing COLDSTART's for all of our tests (which mean you aren't really testing a lot of FATES). @rgknox thought he could do that for a f10 resolution case. Some test mods could inherent from that test mod and you'd have some tests that be cone with either a COLDSTART or an inventory file.

**Comment by glemieux** [comment]
> I also suggest that the Fates base class test-mod should have the COLDSTART setting removed. But, then all other current tests should inherent from the FatesColdDef class which does a COLDSTART, and the names of them all should have FatestColdDef in the name.

It looks like the I may have accidentally included the `./xmlchange CLM_FORCE_COLDSTART="on"` into the base Fates testmod.  I'll need to remove this.

[  ] Remove `./xmlchange CLM_FORCE_COLDSTART="on"` from Fates testmod

**Comment by glemieux** [comment]
> We talked about which FATES test-mods shouldn't be in the aux_clm test list and these are the ones we came up with: FatesNoFire, FatesPPhys, FatesShortList, FatesPRT2 (until changes start to come in on the CTSM side to support it), FatesST3. It looks like FatesFire can just be removed as it's the default.

We talked about this in the context of replacing the clm45 compset test in `aux_clm` on cheyenne.  @ekluzek do you think we should add a subset of these to izumi as well?

**Comment by glemieux** [comment]
> It does seem like you should at least have a NCK test in the fates test suite (addressing #514).

After discussion, Charlie, Ryan and I decided to close #514, so I won't be addressing it with this issue or the associated PR.

**Comment by rgknox** [comment]
Re: what to do about cold-start naming conventions.  I don't have strong feelings on this, but starting a simulation "from nothing", ie cold, is really our only setting for a fates run, until we develop tests with a regional/global inventory initialization.  Maybe there are nuances to what starting cold means.  Having said that, I just re-read the documentation for ```./xmlchange CLM_FORCE_COLDSTART="on"```  .  This just means that finidat='' right?

**Comment by ekluzek** [comment]
Yes the CLM_FORCE_COLDSTART="on just sets finidat to blank, and hence the initial conditions are arbitrary. It sets temperature to the air temp and isothermal all the way down. Soil moisture is some arbitrary fraction. Stuff like that.

I think the main reason to have it in the name though is to make it obvious and also allow you to build versions that aren't doing a cold start. Since you can build tests mods from other ones it's nice to have things separated out for which function it serves. So you can have all the cold start test mods inherit from the cold testmod, but others parallel to that could inherit from a test mods that has a startup IC file in some form.

**Comment by rgknox** [comment]
this:

> I think the main reason to have it in the name though is to make it obvious ...

I'm on board

**Comment by glemieux** [comment]
> I think the main reason to have it in the name though is to make it obvious and also allow you to build versions that aren't doing a cold start.

Here's the commit in #1275 that changes the naming: https://github.com/ESCOMP/CTSM/pull/1275/commits/c5b6b1e98069f9e14620648cf95c55b9fa68b469
